src/Plugins/LevelDBStore/Plugins/Storage/Snapshot.cs

```diff
@@ -9,67 +9,69 @@
 // Redistribution and use in source and binary forms with or without
 // modifications are permitted.
 
-using Neo.IO.Data.LevelDB;
+using Neo.IO.Storage.LevelDB;
 using Neo.Persistence;
 using System.Collections.Generic;
-using LSnapshot = Neo.IO.Data.LevelDB.Snapshot;
 
 namespace Neo.Plugins.Storage
 {
     internal class Snapshot : ISnapshot
     {
-        private readonly DB db;
-        private readonly LSnapshot snapshot;
-        private readonly ReadOptions options;
-        private readonly WriteBatch batch;
+        private readonly DB _db;
+        private readonly SnapShot _snapshot;
+        private readonly WriteBatch _batch;
+        private readonly ReadOptions _readOptions;
+
+        private readonly object _lock = new();
 
         public Snapshot(DB db)
         {
-            this.db = db;
-            snapshot = db.GetSnapshot();
-            options = new ReadOptions { FillCache = false, Snapshot = snapshot };
-            batch = new WriteBatch();
+            _db = db;
+            _snapshot = db.CreateSnapshot();
+            _batch = new();
+            _readOptions = new ReadOptions { FillCache = false, Snapshot = _snapshot };
         }
 
         public void Commit()
         {
-            db.Write(WriteOptions.Default, batch);
+            lock (_lock)
+                _db.Write(_batch);
         }
 
         public void Delete(byte[] key)
         {
-            batch.Delete(key);
+            lock (_lock)
+                _batch.Delete(key);
         }
 
         public void Dispose()
         {
-            snapshot.Dispose();
-        }
-
-        public IEnumerable<(byte[] Key, byte[] Value)> Seek(byte[] prefix, SeekDirection direction = SeekDirection.Forward)
-        {
-            return db.Seek(options, prefix, direction, (k, v) => (k, v));
+            _snapshot.Dispose();
+            _batch.Dispose();
+            _readOptions.Dispose();
         }
 
         public void Put(byte[] key, byte[] value)
         {
-            batch.Put(key, value);
+            lock (_lock)
+                _batch.Put(key, value);
         }
 
-        public bool Contains(byte[] key)
-        {
-            return db.Contains(options, key);
-        }
+        public bool Contains(byte[] key) =>
+            _db.Contains(key, _readOptions);
 
-        public byte[] TryGet(byte[] key)
-        {
-            return db.Get(options, key);
-        }
+        public byte[] TryGet(byte[] key) =>
+            _db.Get(key, _readOptions);
```

Comment:
I think that if we use lock to write we also need it to read, or you could be reading while someone else is writing